cmd/shard_distribution.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Pilosa Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"github.com/pilosa/pilosa/ctl"
+)
+
+var shardDistributioner *ctl.ShardDistributionCommand
+
+func newShardDistributionCommand(stdin io.Reader, stdout, stderr io.Writer) *cobra.Command {
+	shardDistributioner = ctl.NewShardDistributionCommand(stdin, stdout, stderr)
+	shardDistributionCmd := &cobra.Command{
+		Use:   "shard-distribution",
+		Short: "Shard distribution data from pilosa.",
+		Long: `
+Returns the shard distribution based on the provided arguments, or on a running
+Pilosa cluster.
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return shardDistributioner.Run(context.Background())
+		},
+	}
+	flags := shardDistributionCmd.Flags()
+
+	flags.StringVarP(&shardDistributioner.Host, "host", "", "", "host:port of Pilosa")
+	flags.StringVarP(&shardDistributioner.Index, "index", "i", "", "Pilosa index to consider")
```

Comment:
If a non-existent index is specified, we still get a non-error result. Should it return an error?

Reply:
no, we want to be able to ask Pilosa for the shard distribution of any index name

Reply:
Well, then I guess it should return an empty response, like `[]`.

Reply:
I think it should return the shard distribution up to the given max shard

Reply:
What would be the max shard of a non-existent index?

Reply:
you provide the max-shard (default is 0)
the idea being that you could say "if i were to have an index called "foo" with 1000 shards, how would those shards be distributed on a 5-node cluster?" in that case, it's ok that "foo" doesn't exist.